Makefile

```diff
@@ -0,0 +1,382 @@
+# Variables
+APACHE_ENTRY_PATH := $(shell if [ '$(APACHE_BASE_PATH)' = 'main' ]; then echo ''; else echo /$(APACHE_BASE_PATH); fi)
+APP_VERSION := $(shell python -c "print __import__('time').strftime('%s')")
+BASEWAR := print-servlet-2.0-SNAPSHOT-IMG-MAGICK.war
+BRANCH_STAGING := $(shell if [ '$(DEPLOY_TARGET)' = 'dev' ]; then echo 'test'; else echo 'integration'; fi)
+CURRENT_DIRECTORY := $(shell pwd)
+DEPLOYCONFIG ?=
+GIT_BRANCH ?= $(shell git rev-parse --symbolic-full-name --abbrev-ref HEAD)
+HTTP_PROXY := http://ec2-52-28-118-239.eu-central-1.compute.amazonaws.com:80
+INSTALL_DIRECTORY := .venv
+MODWSGI_USER := www-data
+NO_TEST ?= withtests
+NODE_DIRECTORY := node_modules
+PRINT_INPUT := $(BASEWAR) *.yaml *.png WEB-INF
+PRINT_OUTPUT_BASE := /srv/tomcat/tomcat1/webapps/print-chsdi3-$(APACHE_BASE_PATH)
+PRINT_OUTPUT := $(PRINT_OUTPUT_BASE).war
+PRINT_TEMP_DIR := /var/cache/print
+PYTHON_FILES := $(shell find chsdi/* -path chsdi/static -prune -o -type f -name "*.py" -print)
+SHORTENER_ALLOWED_DOMAINS := admin.ch, swisstopo.ch, bgdi.ch
+SHORTENER_ALLOWED_HOSTS :=
+TEMPLATE_FILES := $(shell find -type f -name "*.in" -print)
+USERNAME := $(shell whoami)
+WSGI_APP := $(CURRENT_DIRECTORY)/apache/application.wsgi
+ZADARA_DIR := /var/local/cartoweb/downloads/
+
+# Commands
+AUTOPEP8_CMD := $(INSTALL_DIRECTORY)/bin/autopep8
+FLAKE8_CMD := $(INSTALL_DIRECTORY)/bin/flake8
+MAKO_CMD := $(INSTALL_DIRECTORY)/bin/mako-render
+NOSE_CMD := $(INSTALL_DIRECTORY)/bin/nosetests
+PIP_CMD := $(INSTALL_DIRECTORY)/bin/pip
+PSERVE_CMD := $(INSTALL_DIRECTORY)/bin/pserve
+PYTHON_CMD := $(INSTALL_DIRECTORY)/bin/python
+SPHINX_CMD := $(INSTALL_DIRECTORY)/bin/sphinx-build
+
+# Linting rules
+PEP8_IGNORE := "E128,E221,E241,E251,E272,E501,E711"
+
+# E128: continuation line under-indented for visual indent
+# E221: multiple spaces before operator
+# E241: multiple spaces after ':'
+# E251: multiple spaces around keyword/parameter equals
+# E272: multiple spaces before keyword
+# E501: line length 79 per default
+# E711: comparison to None should be 'if cond is None:' (SQLAlchemy's filter syntax requires this ignore!)
+
+# Colors
+RESET := $(shell tput sgr0)
+RED := $(shell tput setaf 1)
+GREEN := $(shell tput setaf 2)
+
+# Versions
+# We need GDAL which is hard to install in a venv, modify PYTHONPATH to use the
+# system wide version.
+PYTHON_VERSION := $(shell python --version 2>&1 | cut -d ' ' -f 2 | cut -d '.' -f 1,2)
+PYTHONPATH ?= .venv/lib/python${PYTHON_VERSION}/site-packages:/usr/lib64/python${PYTHON_VERSION}/site-packages
+
+.PHONY: help
+help:
+	@echo "Usage: make <target>"
+	@echo
+	@echo "Possible targets:"
+	@echo "- all                Build the app"
+	@echo "- user               Build the user specific version of the app"
+	@echo "- dev                Build the dev version of the app"
+	@echo "- int                Build the int version of the app"
+	@echo "- prod               Build the productive version if the app"
```

Comment:
I guess we can remove it from the help because I just run it to see what happens with it and obviously I create a conflict between apache confs.

and then I wanted to revert back but `make user` doesn't do a `make clean` . If it's not in the help nobody will use it.